code/modules/atmospherics/machinery/datum_pipeline.dm

```diff
@@ -213,8 +213,12 @@ var/pipenetwarnings = 10
 		GL += P.return_air()
```

Comment:
@RemieRichards The problem is here, the null is getting added to the list that's being looped over. It happens when you have a valve attached to a pipenet with nothing on the other side